### PR TITLE
Add GO backend for resiliency scenario

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: http/resiliency/src/test/resources/source_files/circuitbreaker-with-retry-test/src/backend_service/commons.bal [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+int count1 = 0;
+int count2 = 0;
+
+string http1ServicePrefix = "[Http1Service] ";
+string http2ServicePrefix = "[Http2Service] ";
+
+function sendNormalResponse(http:Caller caller, http:Response response, string prefix, int count) {
+    string message = prefix + "OK response. Backend request count: " + count.toString();
+    response.setPayload(message);
+    var result = caller->respond(response);
```
Comment (Manuri) [inline http/resiliency/src/test/resources/source_files/circuitbreaker-with-retry-test/src/backend_service/commons.bal]: Shall we print a log for the error case?

File: http/resiliency/src/test/resources/source_files/circuitbreaker-with-retry-test/src/backend_service/commons.bal [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+function sendErrorResponse(http:Caller caller, http:Response response, string prefix, int count) {
+    response.statusCode = 501;
+    response.setPayload(prefix + "Internal error occurred. Backend request count: " + count.toString());
+    var result = caller->respond(response);
```
Comment (Manuri) [inline http/resiliency/src/test/resources/source_files/circuitbreaker-with-retry-test/src/backend_service/commons.bal]: Shall we print a log for the error case?

File: scenarios-commons/src/main/java/org/ballerinalang/scenario/test/common/http/HttpsClientRequest.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Manuri) [inline scenarios-commons/src/main/java/org/ballerinalang/scenario/test/common/http/HttpsClientRequest.java]: Year should be corrected?

Comment (ThisaruGuruge) [reply scenarios-commons/src/main/java/org/ballerinalang/scenario/test/common/http/HttpsClientRequest.java]: Done

**Pull request merged**